File: 2_analysis/source/analyze_data.py
import pandas as pd
import numpy as np
import statsmodels.api as sm
import matplotlib.pyplot as plt

# =============================================================================

def main():
  mpg_clean = pd.read_csv("../input/mpg.csv")
  regression_table(mpg_clean)
  city_figure(mpg_clean)
  hwy_figure(mpg_clean)

def regression_table(data): 
  reg_cty = sm.OLS.from_formula('displ ~ cty', data).fit()
  print(reg_cty.summary())

  reg_hwy = sm.OLS.from_formula('displ ~ hwy', data).fit()
  print(reg_hwy.summary())

  reg_hwy_cty = sm.OLS.from_formula('displ ~ cty + hwy', data).fit()
  print(reg_hwy_cty.summary())

  latex_table = reg_hwy_cty.summary().as_latex()
  with open("../output/table_reg.tex", "w") as f:
    f.write(latex_table)

def hwy_figure(data):
  plt.scatter(data['displ'], data['hwy'], c=data['year'])
  plt.xlabel("Engine displacement (L)")
  plt.ylabel("Highway fuel economy (mpg)")
  plt.savefig("../output/figure_hwy.jpg")

def city_figure(data):
  plt.scatter(data['displ'], data['cty'], c=data['year'])
  plt.xlabel("Engine displacement (L)")
  plt.ylabel("City fuel economy (mpg)")
  plt.savefig("../output/figure_city.jpg")

# Execute
if __name__ == "__main__":
  main()

File: 2_analysis/source/analyze_data.r
# =============================================================================
# Short description of script's purpose
# =============================================================================

library(tidyverse)
library(stargazer)
library(ggplot2)

# Paths
input_dir <- "../../1_data/output"
output_dir <- "../output"

# =============================================================================

main <- function() {
  load(file.path(input_dir, "mpg.Rdata"))
  regression_table(mpg_clean)
  city_figure(mpg_clean)
  hwy_figure(mpg_clean)
}

regression_table <- function(data) {
  reg_cty <- lm(displ ~ cty, data = data)
  summary(reg_cty)

  reg_hwy <- lm(displ ~ hwy, data = data)
  summary(reg_hwy)

  reg_hwy_cty <- lm(displ ~ hwy + cty, data = data)
  summary(reg_hwy_cty)

  stargazer(reg_cty, reg_hwy, reg_hwy_cty, title = "Results", align = TRUE,
            dep.var.labels = c("Engine displacement (L)"),
            covariate.labels = c("City fuel economy (mpg)",
                                 "Highway fuel economy (mpg)"),
            out = file.path(output_dir, "table_reg.tex"))
}

city_figure <- function(data) {
  p <- ggplot(data, aes(x = displ, y = cty, color = year)) +
    geom_point() +
    xlab("Engine displacement (L)") +
    ylab("City fuel economy (mpg)")
  ggsave(file.path(output_dir, "figure_city.jpg"), plot = p)
}

hwy_figure <- function(data) {
  p <- ggplot(data, aes(x = displ, y = hwy, color = year)) +
    geom_point() +
    xlab("Engine displacement (L)") +
    ylab("Highway fuel economy (mpg)")
  ggsave(file.path(output_dir, "figure_hwy.jpg"), plot = p)
}

# Execute
main()
